proxy/tests/support/client.rs

```diff
@@ -102,6 +102,7 @@ fn run(addr: SocketAddr, version: Run) -> Sender {
                     .build(&reactor);
                 Box::new(rx.for_each(move |(req, cb)| {
                     let mut req = hyper::Request::from(req.map(|()| hyper::Body::empty()));
+                    assert!(req.body_mut().take().unwrap().is_empty());
```

Comment:
I have no idea what this change means. I think the person who reviewed the `http` crate change needs to review this.

Reply:
This is in the test code. The test client currently don't have any way of setting a body, so it's just asserting that it is actually empty.

Reply:
OK, gotcha. I spent too much time misunderstanding this. So, the actual bug fix was 100% in the http crate?

Reply:
Oh, never mind that last comment.